src/runner/settings_window.cpp

```diff
@@ -206,6 +206,37 @@ BOOL run_settings_non_elevated(LPCWSTR executable_path, LPWSTR executable_args,
     return process_created;
 }
 
+// This variable is set based on the os version
+bool use_old_settings = true;
+const int RS5_VERSION = 17999;
+
+// This function obtains the os build number from the registry and accordingly sets the settings executable
+void os_detection()
```

Comment:
Instead of setting a global variable, make `os_detection` return a value. And rename it so it does auto explain what it returns.